src/NuGet.Clients/NuGet.VisualStudio/SolutionRestoreManager/IVsProjectRestoreInfo3.cs

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+#nullable enable
+
+using System.Collections.Generic;
+
+namespace NuGet.SolutionRestoreManager
+{
+    /// <summary>
+    /// Contains project metadata needed for project restore operation.
+    /// </summary>
+    public interface IVsProjectRestoreInfo3
+    {
+        /// <summary>
+        /// The MSBuildProjectExtensionsPath of the project (originally BaseIntermediateOutputPath was used,
+        /// but rather than create a new interface, we changed the meaning of this property).
+        /// </summary>
+        string BaseIntermediatePath { get; }
+
+        /// <summary>
+        /// Target frameworks metadata.
+        /// </summary>
+        IReadOnlyList<IVsTargetFrameworkInfo4> TargetFrameworks { get; }
+
+        /// <summary>
+        /// Collection of tool references.
+        /// </summary>
+        IReadOnlyList<IVsReferenceItem2>? ToolReferences { get; }
+
+        /// <summary>
+        /// Original raw value of TargetFrameworks property as set in a project file.
+        /// </summary>
+        string? OriginalTargetFrameworks { get; }
```

Comment:
The `PackageSpec` class has a property `OriginalTargetFrameworks` which gets populated by this. We could generate an equivalent list by iterating each `TargetFramework`. If you have an opinion, please share. Otherwise, I'll think about it while I'm addressing all the other feedback.

Reply:
We also use this property to set a `CrossTargeting` property in the `PackageSpec`, even if the project's `TargetFrameworks` only contains a single framework.

Therefore, there's no other way to reproduce this behaviour, except to keep this `OriginalTargetFrameworks` property.

Reply:
You are right. 

OriginalTargetFrameworks tells us whether `TargetFramework` or `TargetFrameworks` was used to define the frameworks. 

Thanks for digging into this.